### ISSUE TITLE
The `prefer-add-event-listener` rule should not apply to Koa's `app.context.onerror`

### ISSUE THREAD

**Issue by niftylettuce** [sindresorhus/eslint-plugin-unicorn#174]

**Comment by sindresorhus** [comment]
The only workaround I can think of is to detect `koa` being imported and not report.

It's easy to think we can just detect `app.context.onerror`, but it can be written in so many ways with variables in between and it's just too complicated with the limited things we can detect and do with ESLint.

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $40.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/174)

**Comment by medusalix** [comment]
I would like to work on this issue. If I understood correctly, `koa` and `sax` should be excluded from this rule, right?

**Comment by niftylettuce** [comment]
@medusalix 	those are the two that are most urgent right now, thank you

**Comment by medusalix** [comment]
I thought of doing it like this:

```js
Program() {
    disabled = false;
},

'CallExpression[callee.name="require"]': node => {
    // If 'package' in ['koa', 'sax'] => disabled = true
}

// Disable other selectors based on 'disabled' value
```

Although I already got it working this way, I wanted to ask if some of you would know a better approach to this problem?

**Comment by sindresorhus** [comment]
@medusalix Yeah, that's what I would do too. It also needs to handle `import` statement.

> If 'package' ends with 'koa' or 'sax'

Did you means `ends` => `matches`?

---

And `disabled` => `isDisabled`.

**Comment by medusalix** [comment]
@sindresorhus I'm actually checking the package name via `endsWith`. Consider the following case:

```js
const sax = require('./lib/sax');
```

Maybe we should split the path on each `/` and check whether the last part equals `sax`?
When using `endsWith`, something like this would still pass 😕

```js
const sax = require('./lib/somethingnotrelatedtosax');
```

Do you think a regex would be a better choice?

**Comment by sindresorhus** [comment]
It should not handle relative paths. Only npm packages (`import koa from 'koa';`). So we'll just have a `Set` of npm package imports that disables the rule. Like: `new Set(['koa', 'sax'])`.

**Comment by medusalix** [comment]
Ok, that makes sense.

**Comment by IssuehuntBot** [comment]
@sindresorhus has rewarded $36.00 to @medusalix. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/174)

- :moneybag: Total deposit: $40.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $4.00
